kubernetes/argo.libsonnet

```
@@ -1,5 +1,44 @@
 local ok = import 'outreach.libsonnet';
-{
+
+// namespace for argocd
+local argocdNamespace = 'argocd';
+
+{ 
+  Application(name): ok._Object('argoproj.io/v1alpha1', 'Application', name, namespace=argocdNamespace) {
+    local this = self,
+
+    namespace_:: error 'namespace_ is required',
+    path_:: error 'path_ is required',
+    repo_:: error 'repo_ is required',
+    initial_revision_:: 'none',
+    repo_name_:: std.split(this.repo_, "/")[std.length(std.split(this.repo_, "/"))-1],
+    source_path_:: std.join("/", std.slice(std.split(this.path_, "/"), 0, std.length(std.split(this.path_, "/"))-1, 1)),
+    env_:: {},
+    spec: {
+      destination: {
+        namespace: this.namespace_,
+        server: 'https://kubernetes.default.svc'
+      },
+      project: 'default',
+      source: {
+        path: this.source_path_,
+        repoURL: this.repo_,
+        [if this.initial_revision_ != '' then 'targetRevision']: this.initial_revision_,
+        plugin: {
+          name: 'kubecfg', 
+          env: ok.envList(this.env_) + [
+            { name: 'TAG', value: this.initial_revision_ },
+            { name: 'NAMESPACE', value: this.namespace_ },
+            { name: 'MANIFESTPATH', value: '/tmp/git@github.com_getoutreach_%(name)s/%(path)s' % { name: this.repo_name_, path: this.path_ } },
+          ],
```

Since we're already using `envList` here it seems like it'd be more readable to _fully_ use it.

```suggestion
          env: ok.envList(this.env_ + {
          	TAG: this.initial_revision_,
          	NAMESPACE: this.namespace_,
          	MANIFESTPATH: '/tmp/git@github.com_getoutreach_%(name)s/%(path)s' % { name: this.repo_name_, path: this.path_ }
          }),
```